testing/unittest/iterator_helpers.h

```diff
@@ -0,0 +1,52 @@
+#pragma once
+
+#include <iterator>
+
+
+// Wraps an existing iterator into a forward iterator,
+// thus removing some of its functionality
+template <typename Iterator>
+struct forward_iterator_wrapper {
+    // LegacyIterator requirements
+    using reference = typename Iterator::reference;
+    using pointer = typename Iterator::pointer;
+    using value_type = typename Iterator::value_type;
+    using difference_type = typename Iterator::difference_type;
+    using iterator_category = std::forward_iterator_tag;
+
```

Comment:
Please also add a `static_assert` verifying that `Iterator` is _at least_ a forward iterator. Something like (untested):

```c++
using base_iterator_category = typename thrust::iterator_traits<Iterator>::iterator_category;
static_assert(
    std::is_convertible<base_iterator_category, std::forward_iterator_tag>::value, 
    "<insert a meaningful message goes here>");
```